### PR TITLE
Change Proctitle to IDLE after an Error

### LINKED ISSUE #7891: Incorrect Title for Failed Tasks
### What is the problem?
When a task fails, the process title of the failed task remains. This appears in both the dashboard and to the OS (`htop`).

*Ray version and other system information (Python version, TensorFlow version, OS):*
Python: 3.7
OS: MacOS 10.15.4
Ray: Latest Master

### Reproduction (REQUIRED)
```
import ray
ray.init()
@ray.remote
def fail():
    import NonRealModule
fail.remote()
```

Then run `htop` and "`__main__.fail()`" will remain even after the function errors and returns. 

If we cannot run your script, we cannot fix your issue.

- [x] I have verified my script runs in a clean environment and reproduces the issue.
- [x] I have verified the issue also occurs with the [latest wheels](https://ray.readthedocs.io/en/latest/installation.html).

### PR DESCRIPTION
**Pull request by ijrsvt**
## Why are these changes needed?
If a task errors, its proctitle remains until another task is run. 

## Related issue number
Closes #7891

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://ray.readthedocs.io/en/latest/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failure rates at https://ray-travis-tracker.herokuapp.com/.
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested (please justify below)

### CODE REVIEW

**Review by rkooo567 — APPROVED** [review]
Is there any good way to test this?

**Comment by ijrsvt** [comment]
@rkooo567 I'm not sure what a good way to robustly test this is--mostly because the problems are more UI related.

**Review by simon-mo — APPROVED** [review]

**Comment by simon-mo** [comment]
Let's wait until the master is green to retest and merge this. There might be some tests that check proctitle, just to be sure.

**Pull request merged**
